src/main/java/org/influxdb/dto/Point.java

```diff
@@ -286,22 +286,26 @@ private StringBuilder concatenateFields() {
 		numberFormat.setMinimumFractionDigits(1);
 
 		for (Entry<String, Object> field : this.fields.entrySet()) {
-			sb.append(KEY_ESCAPER.escape(field.getKey())).append("=");
 			loops++;
 			Object value = field.getValue();
-			if (value instanceof String) {
-				String stringValue = (String) value;
-				sb.append("\"").append(FIELD_ESCAPER.escape(stringValue)).append("\"");
-			} else if(useInteger && (value instanceof Integer || value instanceof BigInteger || value instanceof Long)) {
-				sb.append(value).append("i");
-			} else if (value instanceof Number) {
-				sb.append(numberFormat.format(value));
-			} else {
-				sb.append(value);
-			}
 
-			if (loops < fieldCount) {
-				sb.append(",");
+			// Avoiding invalid null type
+			if (value != null) {
```

Comment:
I think it is more readable if you continue if value == null, it also leads to less idented code. 
If you mind change it this way i will merge